Simplified/NYPLSettingsAccountDetailViewController.m

```diff
@@ -1481,18 +1488,27 @@ - (UITableViewCell *)tableView:(__attribute__((unused)) UITableView *)tableView
         if (barcodeImage) {
           self.barcodeImageView = [[UIImageView alloc] initWithImage:barcodeImage];
           self.barcodeImageLabel = [[UILabel alloc] init];
+          self.barcodeTextLabel = [[UILabel alloc] init];
+          self.barcodeTextLabel.text = self.selectedUserAccount.barcode;
```

Comment:
hmm one problem here is that the `barcode` could actually be a username in some situations. I wonder if we should use the `authorizationIdentifier` instead? there should be some comments about this in the code base, but I don't recall exactly where.

Reply:
check out the documentation for `NYPLUserAccount::authorizationIdentifier` and `NYPLUserAccount::barcode`  -- perhaps we can write a method in NYPLSignInBusinessLogic that tries to return the barcode for NYPL at least? I suppose it depends how much we care, and this is a question for Risa ( @killerpuppytails ). But if we just use `barcode` this can essentially display an actual barcode or a username, and the app wouldn't know which one it is.

Reply:
You are right, the `barcode` could be a username.

But there is a note on `autherizationIdentifier` saying that is for internal only and do not expose to public.

Reply:
I just double checked -- no, that is referred to the ILS ID, but that's something different. The main point there is that only for NYPL we can say for certain that the authorizationIdenfier is a barcode.

Reply:
Then should we keep using `barcode`? and only display it if the `barcode` is number only and 14 digits? Although it can still be an username

Reply:
So while we would *like* to show the NYPL patron's barcode even if they only enter a username*, it's not required and you can hide the "Show barcode" feature if it's just a username.

* as they might not remember their barcode and it is used exclusively on some databases